yew/src/virtual_dom/vtag.rs

```diff
@@ -110,6 +118,63 @@ impl Clone for VTag {
     }
 }
 
+#[cfg(feature = "ssr")]
+impl ToHtmlString for VTag {
+    fn to_html_string(&self) -> String {
+        let mut parts: Vec<String> = Vec::new();
+        let tag_name = htmlescape::encode_minimal(&self.tag).to_lowercase();
+        parts.push(format!("<{}", tag_name).to_string());
+
+        for (key_unclean, value) in &self.attributes {
+            let key = key_unclean.to_lowercase();
+            // checked, value (special if textarea), disabled, href?, selected,
+            // kind -> type if input, disallow ref, disallow LISTENER_SET, class
+
+            let mut is_valid = true;
+            for c in key.chars() {
+                let is_alnum = (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+                if !is_alnum && c != '-' && c != '_' && c != ':' {
+                    is_valid = false;
+                    warn!("Invalid attribute name: `{}`", &key);
```

Comment:
There's technically a failure condition here too, so maybe having ToHtmlString::to_html_string return a `Result<String, Err>` would be better?